scripts/abi_check.py

```diff
@@ -371,46 +379,12 @@ def run_main():
             help="output only the list of issues to stdout, instead of a full report",
         )
         abi_args = parser.parse_args()
-        if len(abi_args.old_rev) == 1:
-            old_repo = None
-            old_rev = abi_args.old_rev[0]
-        elif len(abi_args.old_rev) == 2:
-            old_repo = abi_args.old_rev[0]
-            old_rev = abi_args.old_rev[1]
-        else:
-            raise Exception("Too many arguments passed for old version")
-        if len(abi_args.new_rev) == 1:
-            new_repo = None
-            new_rev = abi_args.new_rev[0]
-        elif len(abi_args.new_rev) == 2:
-            new_repo = abi_args.new_rev[0]
-            new_rev = abi_args.new_rev[1]
-        else:
-            raise Exception("Too many arguments passed for new version")
-        old_crypto_repo = None
-        old_crypto_rev = None
-        if abi_args.old_crypto_rev:
-            if len(abi_args.old_crypto_rev) == 1:
-                old_crypto_rev = abi_args.old_crypto_rev[0]
-            elif len(abi_args.old_crypto_rev) == 2:
-                old_crypto_repo = abi_args.old_crypto_rev[0]
-                old_crypto_rev = abi_args.old_crypto_rev[1]
-            else:
-                raise Exception("Too many arguments passed for old crypto version")
-        new_crypto_repo = None
-        new_crypto_rev = None
-        if abi_args.new_crypto_rev:
-            if len(abi_args.new_crypto_rev) == 1:
-                new_crypto_rev = abi_args.new_crypto_rev[0]
-            elif len(abi_args.new_crypto_rev) == 2:
-                new_crypto_repo = abi_args.new_crypto_rev[0]
-                new_crypto_rev = abi_args.new_crypto_rev[1]
-            else:
-                raise Exception("Too many arguments passed for new crypto version")
         abi_check = AbiChecker(
-            abi_args.report_dir, old_repo, old_rev, old_crypto_rev,
-            old_crypto_repo, new_repo, new_rev, new_crypto_rev, new_crypto_repo,
-            abi_args.keep_all_reports, abi_args.brief, abi_args.skip_file
+            abi_args.report_dir, abi_args.old_repo, abi_args.old_rev,
+            abi_args.old_crypto_rev, abi_args.old_crypto_repo,
+            abi_args.new_repo, abi_args.new_rev, abi_args.new_crypto_rev,
+            abi_args.new_crypto_repo, abi_args.keep_all_reports,
+            abi_args.brief, abi_args.skip_file
```

Comment:
That is so much simpler to parse.

Reply:
Given that I'll need to rebase to fix commit messages, would it be worth rebasing this commit into the two commits that added the repository selections initially?

Reply:
That'd make reviewing commit by commit easier.

Reply:
This is optional to do.